allennlp/data/dataset_readers/visual_entailment.py

```diff
@@ -0,0 +1,102 @@
+from allennlp.data.dataset_readers.vision_reader import VisionReader
+
+from allennlp.common.file_utils import json_lines_from_file
+
+logger = logging.getLogger(__name__)
+
+
+@DatasetReader.register("visual-entailment")
+class VisualEntailmentReader(VisionReader):
+    """
+    The dataset reader for visual entailment.
+    """
+
+    @overrides
+    def _read(self, file_path: str):
+        lines = json_lines_from_file(file_path)
+        info_dicts: List[Dict] = list(self.shard_iterable(lines))  # type: ignore
+
+        if not self.skip_image_feature_extraction:
+            # It would be much easier to just process one image at a time, but it's faster to process
+            # them in batches. So this code gathers up instances until it has enough to fill up a batch
+            # that needs processing, and then processes them all.
+            processed_images = self._process_image_paths(
+                [self.images[info_dict["Flickr30K_ID"] + ".jpg"] for info_dict in info_dicts]
+            )
+        else:
+            processed_images = [None for i in range(len(info_dicts))]  # type: ignore
+
+        attempted_instances_count = 0
+        failed_instances_count = 0
+        for info_dict, processed_image in zip(info_dicts, processed_images):
+            hypothesis = info_dict["sentence2"]
+            answer = info_dict["gold_label"]
+
+            instance = self.text_to_instance(processed_image, hypothesis, answer)
+            attempted_instances_count += 1
+            if instance is None:
+                failed_instances_count += 1
+            else:
+                yield instance
+
+            if attempted_instances_count % 2000 == 0:
+                failed_instances_fraction = failed_instances_count / attempted_instances_count
+                if failed_instances_fraction > 0.1:
+                    logger.warning(
+                        f"{failed_instances_fraction*100:.0f}% of instances have no answers."
+                    )
+
+    @overrides
+    def text_to_instance(
+        self,  # type: ignore
+        image: Union[str, Tuple[Tensor, Tensor]],
+        hypothesis: str,
+        label: Optional[str] = None,
+        *,
+        use_cache: bool = True,
+    ) -> Optional[Instance]:
+
+        tokenized_hypothesis = self._tokenizer.tokenize(hypothesis)
+        hypothesis_field = TextField(tokenized_hypothesis, None)
+
+        from allennlp.data import Field
+
+        fields: Dict[str, Field] = {"hypothesis": hypothesis_field}
+
+        if image is not None:
+            if isinstance(image, str):
+                features, coords = next(self._process_image_paths([image], use_cache=use_cache))
+            else:
+                features, coords = image
+
+            fields["box_features"] = ArrayField(features)
+            fields["box_coordinates"] = ArrayField(coords)
+
+        if label:
+            if label == "-":
+                # No gold label could be decided.
+                return None
```

Comment:
This can never be `True`.

Reply:
Ah, I did so based on this sentence in the paper: `

> Specifically, each (Ptext, Htext) pair is
assigned a gold label, indicating the label is agreed by a
majority of crowdsourcing workers (at least 3 out of 5). If
such a consensus is not reached, the gold label is marked as
“-”.

It looks like those aren't actually part of the final dataset, though.

Reply:
Removed.